Tailviewer/BusinessLogic/ActionCenter/OpenFolderAction.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using log4net;
+using Tailviewer.BusinessLogic.FileExplorer;
+using Tailviewer.Core;
+
+namespace Tailviewer.BusinessLogic.ActionCenter
+{
+	public sealed class OpenFolderAction
+		: IOpenFolderAction
+			, IProgress<Percentage>
+	{
+		private static readonly ILog Log = LogManager.GetLogger(MethodBase.GetCurrentMethod().DeclaringType);
+		private Task _task;
+		private readonly IFileExplorer _fileExplorer;
+
+		public OpenFolderAction(string path, IFileExplorer fileExplorer)
+		{
+
+			if (string.IsNullOrEmpty(path))
+				throw new ArgumentNullException(nameof(path));
+
+			if(fileExplorer == null)
+				throw new ArgumentNullException(nameof(fileExplorer));
+
+			_fileExplorer = fileExplorer;
+
+			_task = Task.Run(() =>
+			{
+				if (!FileEx.Exists(path, TimeSpan.FromSeconds(1)))
+				{
+					FullFoldername = FileEx.FindClosestExistingFolder(path);

```

Comment:
If the network drive / server doesn't exist, then an explorer window is still opened in this case.
Instead, the action center should open and an appropriate error message should be displayed to the user.
In order to do so, you need to subclass AbstractNotificationViewModel and construct it inside ActionCenterViewModel.AddNotification(). Also add a DataTemplate for this new view model (you can take ExportViewModel.cs and ExportTemplate.xaml as an example) and include its resource dictionary inside ActionCenterControl.xaml.

Reply:
I thought about it a lot and I would argue, that we don't need it. The reason is, if there is some kind of trouble with the file, we're already getting a visual clue. We're constantly checking if the file is actually there. Opening action center wouldn't provide any additional helpful information.
![image](https://user-images.githubusercontent.com/10827743/34405093-9747e186-ebb1-11e7-9e67-d6dba33451cc.png)
On the other hand, launching the explorer might help. For example, if the network drive isn't mapped, we can immediately map it in explorer. If the server doesn't respond/exist, well... maybe this case is special. But still we are better off with explorer, for we can turn network discovery on, explore network devices and so on.

Reply:
Agreed.